source/frontend/src/features/mapSideBar/management/tabs/activities/edit/models.ts

```diff
@@ -82,17 +85,29 @@ export class ManagementActivityFormModel {
             ...getEmptyBaseAudit(0),
           };
         }),
-      activityProperties: this.activityProperties
-        .filter(exists)
-        .map<ApiGen_Concepts_PropertyActivityProperty>(x => x.toApi()),
+      activityProperties: this.selectedProperties.map(x => {
+        const matched =
+          this.activityProperties.filter(y => y.propertyId === x.propertyId)[0] ?? null;
```

Comment:
it seems you are trying to find a single element in the array - why not use `Array.find()` instead of `filter[0]`?